### DISCUSSION TITLE
keyball61一部のキーが反応しない

### DISCUSSION THREAD

**Discussion by smith0120** [Yowkees/keyball#404] [Q&A] [answered]
はんだ初心者で頑張ってみましたが、添付のように一部のキーが反応しません。
過去のQ&Aなども参考に何度かつけ直しましたが変わらずの状況です。すべてピンセットでショートさせても反応がないのでダイオードなのでしょうか？一部は取り替えたのですが変わらずなので、少しでもヒント頂けるとうれしいです。

左：１、１１、３７（追加：４、１８、１４、１９、２４，２９、３５、３６、３７）
右：２４、２８、３２

→１１番の右側ランドはもしかすると取れてしまっているかもしれません。その際の対処も教えていだけますでしょうか？

※左側はProMicroのピンソケットを裏表逆につけてしまい、つけ直しているのでそれも気にはな
ります。。。

![IMG_0359](https://github.com/Yowkees/keyball/assets/46736502/e6e82519-b495-47aa-a30a-1165f34c1408)
![IMG_0360](https://github.com/Yowkees/keyball/assets/46736502/81218b54-c3e8-4968-8b3d-e23e908a4dae)

**Comment by smith0120** [comment]
改めて、左右を繋げて確認したところなんだか左の駄目な箇所がかなり増えてしまいました。
→写真を差し替えさせてもらいました。
あと、左側にUSBを接続すると左側の配列がそのまま裏返した右手の配列になってしまいます。裏側の「LEFT」はショートさせているのですが。
右側USBに指すと正常です。ダメダメな状況ですみませんが、合わせてアドバイスお願いできますでしょうか？

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ProMicroを半田付けしなおしているということで、ProMicro起因の問題を切り分けるため、
左右のProMicroを入れ替えて挙動がどうなるかもご確認お願いします。

【左側】
左右逆の配列となるのは、LEFT判定キーにも不具合があるためと思われますので、ひとまず気にしなくて大丈夫です。
まず最下段のキーを確認したいと思います。
ProMicro裏にROW4の刻印がある端子がありますが、これが他の端子とショートしていないかご確認ください。
またROW4の端子がミドル基板とProMicro本体ときちんと導通しているかもご確認ください（ピンソケットの不具合の確認）

【右側】
下記画像の青丸で示したダイオードが不具合キーに該当します。
これらの極性や半田付けを見直してみてください。
また、左側と同様にROW2の端子（緑点）が他の端子とショートしていないかもご確認ください。

以上、お手数ですがご確認お願いいたします。
![KB61R_keyfix1](https://github.com/Yowkees/keyball/assets/84209772/1870fa21-c36b-4547-87a1-76b96e4b5810)

**Reply by smith0120** [reply]
付け間違えたのが「ProMicro用ピンヘッダ」だけなのですが外すときに壊れてしまい、ProMicroに付属しているコンスルーで行ったのですが基盤側にもはんだしてしまい。。。
直ぐには交換出来ないでのすみませんが教えて頂いた内容を根気よく対応させてもらいます。

**Reply by smith0120** [reply]
まだ途中ですが、少しわかった部分がありますので何度もすみませんがご教授ください。

【左側】
現状ではショートしていることは確認出来ていませんが、全体的な見直しで少し回復している箇所があります。
聞いてばかりですみませんが、1と4のキーのダイオード箇所を教えて頂けますでしょうか？

【右側】
こちらも現状ではショートしていることは確認出来ていませんが、USB接続した際に「k,j,h」が自動で入力されてしまうことがわかりました。ダイオードは半田しなおしましたが復旧せずの状態です。
何かしらヒントになるでしょうか？


![左](https://github.com/Yowkees/keyball/assets/46736502/71cab51b-cf42-47b2-b1dd-73b263f92791)
![右](https://github.com/Yowkees/keyball/assets/46736502/cc1eba08-ff26-4b7c-87a5-026e84541764)

**Comment by Yowkees** [comment, marked as answer]
ご対応ありがとうございます。

Keyballシリーズは下記にて回路図と基板データを公開しておりますので、KiCAD等のフリーソフトで表示していただくと、線を辿って配線を確認いただけますのでご参考までに。
https://github.com/Yowkees/keyball/tree/main/keyball61/design_data

【左側】
下記画像の緑丸が該当するダイオードになります。その他、キーソケット自体の半田付けミスも非常に多いです。キーソケットの端子とミドル基板の間にきちんと半田が浸み込んでいるかご確認ください。
![KB61R_eachkeyfix](https://github.com/Yowkees/keyball/assets/84209772/11efbbf8-0fb0-4316-915a-7f9cd849844b)

【右側】
ダイオードの極性確認と交換で直らないということで、ProMicroのピンヘッダーに原因がありそうです。
ProMicroを交換できた時点でご確認をお願いします。

なお、新しい12Pピンソケットとピンヘッダーはキット購入者様限定で白銀ラボウェブショップからも販売可能ですので、ご希望の場合はお問合せよりご連絡いただければ幸いです。

以上、よろしくお願いいたします。

**Reply by smith0120** [reply]
ご丁寧にありがとうございます。
KiCADでも確認させてもらいました。とりあえずProMicroの交換を検討させてもらいます。

**Reply by smith0120** [reply]
色々と試行錯誤して、右側はProMicroのところでショート、左側はショートは確認出来ず諦めかけていましたがKiCADで配線を追ってショートしてなんとか完動まで行きました。
これでkeyballユーザ始めれます。ありがとうございました。
